lib/Dialect/TritonGPU/Transforms/Utility.cpp

```diff
@@ -133,7 +133,7 @@ unsigned getNumElementsPerThread(Operation *op, SmallVector<unsigned> order,
   unsigned maxMultipleBytes = valInfo.getDivisibility(order[0]);
   unsigned maxMultiple = std::max(maxMultipleBytes / elemNumBytes, 1u);
   unsigned maxContig =
-      std::min(valInfo.getContiguity(order[0]), shapePerCTA[order[0]]);
+      std::min(valInfo.getContiguity(order[0]), shape[order[0]]);
```

Comment:
How does this make sense?

Reply:
Imagine the contiguous dim size is 4 but it's split across 4 CTAs. This function would pick 4 elements per thread, even though we are only loading a single element in each CTA. 

Either this would create an invalid layout, or it would result in too much broadcasting.